viewing_party/party.py

```diff
@@ -1,23 +1,206 @@
+
+
+
+# # -----------------------------------------
+# # ------------- WAVE 3 --------------------
+# # -----------------------------------------
+def get_unique_watched(user_data):
+    
+# initiated two emtpy list: user and friends
+#we have an outer dictionary, one with "watched" key and one with "friends " key
+# {'watched': [{'title': 'The Lord of the Functions: The Fellowship of the Function', 'genre': 'Fantasy', 'rating': 4.8}, {'title': 'The Lord of the Functions: The Two Parameters', 'genre': 'Fantasy', 'rating': 4.0}, {'title': 'The Lord of the Functions: The Return of the Value', 'genre': 'Fantasy', 'rating': 4.0}, {'title': 'The JavaScript and the React', 'genre': 'Action', 'rating': 2.2}, {'title': 'Recursion', 'genre': 'Intrigue', 'rating': 2.0}, {'title': 'Instructor Student TA Manager', 'genre': 'Intrigue', 'rating': 4.5}],
+# 'friends': [{'watched': [{'title': 'The Lord of the Functions: The Fellowship of the Function', 'genre': 'Fantasy', 'rating': 4.8}, {'title': 'The Lord of the Functions: The Return of the Value', 'genre': 'Fantasy', 'rating': 4.0}, {'title': 'The Programmer: An Unexpected Stack Trace', 'genre': 'Fantasy', 'rating': 4.0}, {'title': 'It Came from the Stack Trace', 'genre': 'Horror', 'rating': 3.5}]}, {'watched': [{'title': 'The Lord of the Functions: The Fellowship of the Function', 'genre': 'Fantasy', 'rating': 4.8}, {'title': 'The JavaScript and the React', 'genre': 'Action', 'rating': 2.2}, {'title': 'Recursion', 'genre': 'Intrigue', 'rating': 2.0}, {'title': 'Zero Dark Python', 'genre': 'Intrigue', 'rating': 3.0}]}]
+#They want to see the what the user saw that none of their friends saw, and return list of movies 
+ 
+    friends_watched_titles = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in friends_watched_titles:
+                friends_watched_titles.append(movie["title"])
+                
+    
         
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+    unique_movies = []
+    for user_watched_movie in user_data["watched"]:
+        if user_watched_movie["title"] not in friends_watched_titles:
+            unique_movies.append(user_watched_movie)
+    return unique_movies
+    
+    
+def get_friends_unique_watched(user_data):
+#get only one of the friend's movie watched list, compare it to the users to see what the friend has saw the user hasn't
+
+
+    user_list =[]
+    friend_unique_movies = []
+    friend_unique_no_duplicate = []
+
+    for movie in user_data["watched"]:
+        user_list.append(movie)
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in user_list:
+                friend_unique_movies.append(movie)
+    #removing duplicates
+    friend_unique_no_duplicate = [i for n, i in enumerate(friend_unique_movies)

```

Comment:
Interesting application of enumerate! Would recommend variable names that are more descriptive so one would more easily know what is represented by `i` and `n`.